src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceVariableService.cs

```diff
@@ -392,5 +393,50 @@ where NodeMatchesExpression(originalSemanticModel, currentSemanticModel, syntaxF
                                                        .Where(p => p.ContainingSymbol.IsAnonymousFunction());
             return anonymousMethodParameters;
         }
+
+        protected static async Task<(SemanticDocument newSemanticDocument, ISet<TExpressionSyntax> newMatches)> ComplexifyParentingStatements(
```

Comment:
I pulled this method up from the C# derived class. The only change is that `ExpressionSyntax` was made generic, using `TExpressionSyntax`.